### DISCUSSION TITLE
Question that the performance bewteen chrome and servo

### DISCUSSION THREAD

**Discussion by Asun0204** [servo/servo#29008] [General]
@jdm Servo's performance is several even ten times slower than chrome. Is something wrong? Or the fact is that servo's performance is worse than chrome.

Test page: wikipage without script tag.

|             | Parse HTML | Parse<br> Stylesheet | Recalculate<br> Style | Layout | Paint | Pre-Paint | Composite |
| --- | --- | --- | --- | --- | --- | --- | --- |
| servo       | 13.93      |                      | 403.35                | 507.26 |       |           | 42.48     |
| chrome      | 21.6       |                      | 30.3                  | 36.2   | 4.2   | 7.9       | 2.5       |
| chrome 2018 | 29.4       | 0.9                  | 57.7                  | 26.2   | 9.2   |           | 1.7       |

> unit: ms
> test platform: MacBook Pro 2018 13-inch macOS 12.5.1
> chrome version: 105.0.5195.125
> chrome 2018 version: 63.0.3239.132
> servo commit: 08bc2d53579c9ab85415d4363888881b91df073b

**Comment by seasonmac** [comment]
Mentioned in your test form, parse html is significantly better than Chrome, is that ture?

**Comment by Asun0204** [comment]
Only wikipedia pages perform better than Chrome, other pages is not. And the build options of --with-layout-2020 could significantly improve performance of Parse HTML.
